packages/gatsby-remark-responsive-iframe/src/index.js

```diff
@@ -1,85 +1,83 @@
 const visit = require(`unist-util-visit`)
 const cheerio = require(`cheerio`)
-const Promise = require(`bluebird`)
 const { oneLine } = require(`common-tags`)
 const _ = require(`lodash`)
 
-const isPixelNumber = n => /\d+px$/.test(n)
-
-const isUnitlessNumber = n => {
-  const nToNum = _.toNumber(n)
-  return _.isFinite(nToNum)
-}
+const needsSemicolon = str => !str.endsWith(`;`)
 
-const isUnitlessOrPixelNumber = n =>
-  n && (isUnitlessNumber(n) || isPixelNumber(n))
+/**
+ * Convert anything to number, except for % value
+ * @param {*} n something to be converted to number
+ * @returns {number}
+ */
+const convert = n =>
+  typeof n === `string` && n.endsWith(`%`) ? NaN : parseInt(n, 10)
```

Comment:
On first reading of this I thought it would be problematic if unit other than `%` or `px` was used - but seems like iframe width/height don't understand other units (like `em`, `rem`, `vw`, etc) and just treats them as `px` - worth adding comment about this, that we don't have to check units other than `%` because of that?

Reply:
That's a good idea, I should probably add that. 

Browser engines parse `width` & `height` the same way as `parseInt(n, 10)`, so actually even `<iframe width="500 this is not invalid">` will return an iframe with 500px width! If it was an invalid string, it'll fallback to the default 300px.

Reply:
Hhhm I goofed here, `n.endsWith(`%`)` doesn't account for `width="100% "`. Will trim that off

Reply:
Nice catch with adding `.trim()`!